src/main/scala/rps/Game.scala

```diff
@@ -1,57 +1,52 @@
 package rps
 
-object Move extends Enumeration {
-  type Move = Value
-  val Rock = Value("0")
-  val Paper = Value("1")
-  val Scissor = Value("2")
+import scala.util.Random
 
-  def isOrderType(s: String) = values.exists(_.toString == s)
+object Moves {
+  sealed class Move(val id: String, val name: String)
+
+  case object Rock extends Move("0", "Rock")
+  case object Paper extends Move("1", "Paper")
+  case object Scissors extends Move("2", "Scissors")
+
+  val moves: Set[Move] = Set(Rock, Paper, Scissors)
 }
 
-object Result extends Enumeration {
-  type Result = Value
-  val Won = Value("you won!")
-  val Lost = Value("you lost...")
-  val Tie = Value("it's a tie!")
+object Results {
+  sealed class Result(val message: String)
```

Comment:
this is a general comment, that you can disregard in this specific case.

In functional programming, we usually define "data classes", meaning classes that have no pre-attached behavior, but that just represent data.

In this case, attaching `message` to a class is partially breaking this, since the message is a concern of a specific behavior (interpreting the result for a user).

Normally, you would have a function that interprets the data and decides what do to with them, for instance

```scala
def resultMessage(r: Result): String = r match {
  case Result.Won => "you won!"
  case Result.Lost => "you lost..."
  case Result.Tie => "it's a tie!"
}
```

So the idea is to attach behaviors to data when needed, instead of pre-defining the behavior  when defining the data.

This is the foundation of ad-hoc polymorphism (aka typeclasses), which is a more advanced topic we'll briefly discuss.

Reply:
This is REALLY interesting. I can't wait to dig a little deeper into the subject 😮